models/DataObject/Classificationstore/Service.php

```diff
@@ -21,18 +21,35 @@
 
 class Service
 {
+    /**
+     * @var array Used for storing definitions
+     */
+    protected static $definitionsCache = [];
+
     /**
      * @param KeyConfig $keyConfig
      *
      * @return DataObject\ClassDefinition\Data
      */
     public static function getFieldDefinitionFromKeyConfig($keyConfig)
     {
+        if ($keyConfig instanceof KeyConfig) {
+            $cacheId = $keyConfig->getId();
+        }
+
+        if ($keyConfig instanceof KeyGroupRelation) {
+            $cacheId = $keyConfig->getGroupId() . "_" . $keyConfig->getKeyId();
```

Comment:
even if it is a relation, the key and therefore the definition is still the same, right ?

Reply:
I think you're right - the definition is taken from KeyConfig table:
pimcore/models/DataObject/Classificationstore/KeyGroupRelation/Dao.php:43

Reply:
I've made the fix.